tools/bridge/tests/test_event_fetcher.py

```diff
@@ -0,0 +1,297 @@
+from gevent import Greenlet, sleep, joinall
+
+from bridge.event_fetcher import EventFetcher
+
+
+@pytest.fixture
+def transfer_event_signature():
+    return "Transfer(address,address,uint256)"
+
+
+@pytest.fixture
+def transfer_event_argument_filter(foreign_bridge_contract):
+    return [None, f"0x000000000000000000000000{foreign_bridge_contract.address[2:]}"]
+
+
+@pytest.fixture
+def foreign_chain_max_reorg_depth():
+    return 10
+
+
+@pytest.fixture
+def transfer_event_fetcher(
+    w3_foreign,
+    token_contract,
+    transfer_event_signature,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    return EventFetcher(
+        web3=w3_foreign,
+        contract_address=token_contract.address,
+        event_signature=transfer_event_signature,
+        event_argument_filter=transfer_event_argument_filter,
+        event_queue=transfer_event_queue,
+        max_reorg_depth=foreign_chain_max_reorg_depth,
+    )
+
+
+@pytest.fixture
+def transfer_tokens_to(token_contract, premint_token_address):
+    def transfer(receiver):
+        token_contract.functions.transfer(receiver, 1).transact(
+            {"from": premint_token_address}
+        )
+
+    return transfer
+
+
+@pytest.fixture
+def transfer_tokens_to_foreign_bridge(foreign_bridge_contract, transfer_tokens_to):
+    def transfer():
+        transfer_tokens_to(foreign_bridge_contract.address)
+
+    return transfer
+
+
+def test_instantiate_event_fetcher_with_non_existing_contract(
+    w3_foreign,
+    transfer_event_signature,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(ValueError):
+        return EventFetcher(
+            web3=w3_foreign,
+            contract_address="0x0000000000000000000000000000000000000000",
+            event_signature=transfer_event_signature,
+            event_argument_filter=transfer_event_argument_filter,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=foreign_chain_max_reorg_depth,
+        )
+
+
+def test_instantiate_event_fetcher_with_not_existing_event_signature(
+    w3_foreign,
+    token_contract,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(ValueError):
+        return EventFetcher(
+            web3=w3_foreign,
+            contract_address=token_contract.address,
+            event_signature="NonExistingEvent(bytes)",
+            event_argument_filter=transfer_event_argument_filter,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=foreign_chain_max_reorg_depth,
+        )
+
+
+def test_instantiate_event_fetcher_with_negative_event_fetch_limit(
+    w3_foreign,
+    token_contract,
+    transfer_event_signature,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(AssertionError):
```

Comment:
I feel you have a misconception about assertion. An assert in the code is an additional safety net to test for something that you believe is true anyways, hence the name assert (german: behaupten). You should not rely on it, and so also not test for it. Asserts can be removed. According to the official documentation the code of an assert is equivalent to:
```
if __debug__:
    if not expression1: raise AssertionError(expression2)
```

So either remove all the tests, if you dont need those assertions, or change them in to raising exceptions.

Reply:
I fail to see the reasoning here. If you want to make sure a certain part of the code raises an assertion, why should you not test for it?

Reply:
For me testing for raising an assertion is showing a wrong assumption about assertions in general. Assertion should never get raised in right code, you should not be able to trigger them in right code, assertion could be removed. So making sure that a certain part of the code raises an assertion does not make sense.

Reply:
Okay so my take-away is to go for `if ... raise` blocks instead of using assertion statements.

Reply:
:heavy_check_mark: